membership-attribute-service/app/models/ExistingPaymentOption.scala

```diff
@@ -0,0 +1,56 @@
+package models
+
+import com.gu.memsub.subsv2.{Subscription, SubscriptionPlan}
+import com.gu.memsub.{GoCardless, PaymentCard, PaymentMethod, Product}
+import com.gu.zuora.rest.ZuoraRestService.ObjectAccount
+import com.typesafe.scalalogging.LazyLogging
+import play.api.libs.json.Json
+import org.joda.time.LocalDate.now
+
+case class ExistingPaymentOption(
+  freshlySignedIn: Boolean,
+  objectAccount: ObjectAccount,
+  paymentMethodOption: Option[PaymentMethod],
+  subscriptions: List[Subscription[SubscriptionPlan.AnyPlan]]
+)
+
+object ExistingPaymentOption {
+
+  implicit class ResultLike(existingPaymentOption: ExistingPaymentOption) extends LazyLogging {
+
+    import existingPaymentOption._
+
+    private def getSubscriptionFriendlyName(plan: SubscriptionPlan.AnyPlan): String = plan.product match {
+      case _: Product.Weekly => "Guardian Weekly"
+      case _: Product.Membership => plan.productName + " Membership"
+      case _: Product.Contribution => plan.name
+      case _ => plan.productName // Newspaper & Digipack
+    }
+
+    private val paymentPart = paymentMethodOption match {
+      case Some(card: PaymentCard) => Json.obj(
+        "card" -> card.paymentCardDetails.map(_.lastFourDigits)
+      )
+      case Some(dd: GoCardless) => Json.obj(
+        "mandate" -> dd.accountNumber
+      )
+      case _ => Json.obj()
+    }
+
+    private val sensitiveDetailIfApplicable = if (freshlySignedIn) {
+      Json.obj(
+        "billingAccountId" -> objectAccount.id.get,
+        "subscriptions" -> subscriptions.map(subscription => Json.obj(
+          "isCancelled" -> subscription.isCancelled,
+          "isActive" -> (!subscription.isCancelled && subscription.termEndDate.isAfter(now)),
```

Comment:
Bit of an edge case, but I think this check might accidentally exclude users with an auto-renewing sub who are due to renew later that day...

Reply:
Note that this doesn't exclude anything, it just sets some booleans in the response which is then used in the render in `support-frontend` to prefix the subscription name with 'current' or 'recent' depending on if `isActive` is true or false, respectively.
https://github.com/guardian/support-frontend/blob/bb12211b026c45d38ce0481ffaea809e4f013045/support-frontend/assets/helpers/existingPaymentMethods/existingPaymentMethods.js#L86

Reply:
My mistake - would probably just lead to a very short-lived copy bug then!

Reply:
yeah which i don't think is a problem, unless you're sugessting `isAfter(now.minusDays(1))` ?

Reply:
in fact DONE in https://github.com/guardian/members-data-api/pull/376/commits/520f4469089204cef9b4cb74fbc18773d4dd4b94

https://github.com/guardian/members-data-api/blob/520f4469089204cef9b4cb74fbc18773d4dd4b94/membership-attribute-service/app/models/ExistingPaymentOption.scala#L45